src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -82,7 +82,55 @@ protected override void Generate(MigrationOperation operation, IModel model, Mig
                 operation = filteredOperation;
             }
 
-            base.Generate(operation, model, builder, false);
+            #region Customized base call
+
+            // TODO: Could EF Core make this easier to override?
```

Comment:
I very much agree...

We could propose adding a `TableTraits()` method much like the `IndexTraits()` which already exists, and which provides an extension point for this kind of thing. Another option which may be better, is to refactor out the generation of the columns (as well as of each individual column) into separate methods. This way the main method is only responsible for the CREATE TABLE and the parentheses...

Go ahead and do it if you feel like it, otherwise let me know and I'll submit it.

Reply:
Sounds good, I'll go ahead and open the issue in a little while.

Reply:
Opened https://github.com/aspnet/EntityFrameworkCore/issues/14374 to suggest a refactor with separate methods.